app/controllers/katello/api/v2/host_tracer_controller.rb

```diff
@@ -16,6 +16,46 @@ def index
       respond_for_index(:collection => collection)
     end
 
+    api :PUT, "/traces/resolve", N_("Resolve Traces")
+    param :trace_ids, Array, :required => true, :desc => N_("Array of Trace IDs")
+    def resolve
+      traces = Katello::HostTracer.where(id: params[:trace_ids])
```

Comment:
We need to ensure that that for the given traces, the user has edit permission for the related hosts. We can use our authorization framework for this.

Create a new module `app/models/katello/authorization/host_tracer.rb`
```
module Katello                                                                                                                                                                                
  module Authorization::HostTracer                                                                                                                                                            
    extend ActiveSupport::Concern                                                                                                                                                             
                                                                                                                                                                                              
    include Authorizable                                                                                                                                                                      
                                                                                                                                                                                              
    module ClassMethods                                                                                                                                                                       
      def resolvable                                                                                                                                                                          
        relation = joins_authorized(::Host::Managed, :edit_hosts)                                                                                                                             
        relation = relation.where(organization_id: User.current.organization_ids) unless User.current.admin?                                                                                  
        relation                                                                                                                                                                              
      end                                                                                                                                                                                     
    end                                                                                                                                                                                       
  end                                                                                                                                                                                         
end
```

This code gives us a new scope: `HostTracer.resolvable` which will return an active record relation of traces we can resolve. If we're an admin it returns everything, otherwise it returns the traces we can edit based on our edit_host permissions.

you can include that module in this class and use `Katello::HostTracer.readable.where.....` here

Reply:
I haven't tested it but this should be 99% of the way there!

Reply:
Added and fixed, will submit to jenkins and see if the same permissions tests fails.